elk/evaluation/evaluate.py

```diff
@@ -1,121 +1,78 @@
-import csv
-import os
-import pickle
 from dataclasses import dataclass
-from functools import partial
-from hashlib import md5
 from pathlib import Path
-from typing import List, Literal, Optional, cast
+from typing import Literal, Optional
 
 import torch
-import torch.multiprocessing as mp
-import yaml
-from simple_parsing.helpers import Serializable, field
-from torch import Tensor
-from tqdm.auto import tqdm
+from simple_parsing import Serializable, field
 
-from datasets import DatasetDict
-from elk.training.preprocessing import normalize
-
-from ..extraction import ExtractionConfig, extract
-from ..files import elk_reporter_dir, memorably_named_dir
-from ..utils import (
-    assert_type,
-    int16_to_float32,
-    select_train_val_splits,
-    select_usable_devices,
-)
+from datasets import Split
+from elk.extraction.extraction import Extract
+from elk.files import elk_reporter_dir
+from elk.run import Run
 
 
 @dataclass
-class EvaluateConfig(Serializable):
-    target: ExtractionConfig
+class Eval(Serializable):
+    """
+    Full specification of a reporter evaluation run.
+
+    Args:
+        data: Config specifying hidden states on which the reporter will be evaluated.
+        source: The name of the source run directory
+            which contains the reporters directory.
+        normalization: The normalization method to use. Defaults to "meanonly". See
+            `elk.training.preprocessing.normalize()` for details.
+        num_gpus: The number of GPUs to use. Defaults to -1, which means
+            "use all available GPUs".
+        debug: When in debug mode, a useful log file is saved to the memorably-named
+            output directory. Defaults to False.
+    """
+
+    data: Extract
```

Comment:
Just for the others from our conversation in the chat:
> Your change requests should probably be maybe implemented as a new feature by itself, since the current pull request is only about refactoring, but we can add them soonish after merging...